cpp/src/arrow/filesystem/filesystem.cc

```diff
@@ -359,20 +360,31 @@ struct FileSystemUri {
 
 Result<FileSystemUri> ParseFileSystemUri(const std::string& uri_string) {
   FileSystemUri fsuri;
-  RETURN_NOT_OK(fsuri.uri.Parse(uri_string));
+  auto status = fsuri.uri.Parse(uri_string);
+  if (!status.ok()) {
+#ifdef _WIN32
+    // Could be a "file:..." URI with backslashes instead of regular slashes.
```

Comment:
Perhaps this fallback should only be used when `uri_string.starts_with("file:")`; backslashes may mean something else for other schemes

Reply:
You're right.